Specify that we recomend typescript over JS for extensions (#8411)
jupyterlab/jupyterlab: docs/source/developer/extension_dev.rst

-  A core Application object
-  Plugins

> Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.
> Tutorials

Reviewer: Oops, I accidentally deleted an important newline

```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.

Tutorials
```

-  A core Application object
-  Plugins

> Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.

Reviewer: ```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in [TypeScript](https://www.typescriptlang.org/), which is used for the JupyterLab core extensions and many popular community extensions.
```

> Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in [TypeScript](https://www.typescriptlang.org/), which is used for the JupyterLab core extensions and many popular community extensions.

Reviewer: ```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in `TypeScript <https://www.typescriptlang.org/>`_, which is used for the JupyterLab core extensions and many popular community extensions.
```

-  A core Application object
-  Plugins

> We recommend writing extensions in Typescript, but technically it doesn't matter what you write them in
> as long as they can be packaged as standard Javascript. 
> 

Reviewer: Good point. How about this language?
```suggestion
Extensions are distributed as JavaScript packages, so you can write extensions in JavaScript or any language that compiles to JavaScript. We recommend writing extensions in TypeScript, which is used for the JupyterLab core extensions and many popular community extensions.
```

Author: Much more eloquent, thank you Jason!
